### PR TITLE
add option to remove hypixel's lobby join messages

### LINKED ISSUE #102: Ability to remove player join messages in hypixel lobbies
https://github.com/AxolotlClient/AxolotlClient-mod/issues/13#issuecomment-1642903972

### PR DESCRIPTION
**Pull request by moehreag**
Currently supported languages:
- english (lobby join + mystery box findings)
- spanish (lobby join)

more languages can also be added via resource packs.

closes #102

### CODE REVIEW

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import net.fabricmc.fabric.api.resource.SimpleSynchronousResourceReloadListener;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.resource.ResourceType;
+import net.minecraft.util.Identifier;
+
+public class HypixelMessages implements SimpleSynchronousResourceReloadListener {
+
+	@Getter()
+	private static final HypixelMessages Instance = new HypixelMessages();
```
Comment (TheKodeToad) [inline 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]:
```suggestion
	private static final HypixelMessages instance = new HypixelMessages();
```
not important but why not this?

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			}).filter(Objects::nonNull).forEach(resource -> {
+				int i = resource.getId().getPath().lastIndexOf("/")+1;
+				String lang = resource.getId().getPath().substring(i, i+5);
+			JsonObject lines = GsonHelper.GSON.fromJson(new InputStreamReader(resource.getInputStream()), JsonObject.class);
+			AxolotlClient.LOGGER.debug("Found message file: "+resource.getId());
+			languageMessageMap.computeIfAbsent(lang, l -> new HashMap<>());
+			Map<String, Pattern> map = languageMessageMap.get(lang);
+			lines.entrySet().forEach(entry -> {
+				Pattern pattern = Pattern.compile(entry.getValue().getAsString());
+				map.computeIfAbsent(entry.getKey(), s -> pattern);
```
Comment (TheKodeToad) [inline 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]:
```suggestion
				map.putIfAbsent(entry.getKey(), pattern);
```

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+				messageLanguageMap.computeIfAbsent(entry.getKey(), s -> new HashMap<>());
+				messageLanguageMap.get(entry.getKey()).put(lang, pattern);
```
Comment (TheKodeToad) [inline 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]:
```suggestion
				messageLanguageMap.computeIfAbsent(entry.getKey(), s -> new HashMap<>()).put(lang, pattern);
```

**Review by TheKodeToad — CHANGES_REQUESTED** [review]

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java (hunk 3 above)
Comment (moehreag) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: .putIfAbsent returns the previous value, not the potentially new one

Comment (TheKodeToad) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: nop

Comment (TheKodeToad) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: image
it returns the same as get would

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+			.map(id -> {
+				try {
+					return manager.getResource(id);
+				} catch (IOException e) {
+					return null;
+				}
+			}).filter(Objects::nonNull).forEach(resource -> {
+				int i = resource.getId().getPath().lastIndexOf("/")+1;
+				String lang = resource.getId().getPath().substring(i, i+5);
+			JsonObject lines = GsonHelper.GSON.fromJson(new InputStreamReader(resource.getInputStream()), JsonObject.class);
```
Comment (DarkKronicle) [inline 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: Indentation is weird here

File: 1.16_combat-6/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+			.map(id -> {
+				try {
+					return manager.getResource(id);
+				} catch (IOException e) {
+					return null;
+				}
+			}).filter(Objects::nonNull).forEach(resource -> {
+				int i = resource.getId().getPath().lastIndexOf("/")+1;
+				String lang = resource.getId().getPath().substring(i, i+5);
+			JsonObject lines = GsonHelper.GSON.fromJson(new InputStreamReader(resource.getInputStream()), JsonObject.class);
```
Comment (DarkKronicle) [inline 1.16_combat-6/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: Indentation is also weird here

**Review by DarkKronicle — COMMENTED** [review]
Looks good after a quick glance. Can't see anything wrong.

File: 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java (hunk 3 above)
Comment (moehreag) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: `putIfAbsent` has a different return value than `computeIfAbsent`, the latter returns the current (new) value while the former returns the previous (older) value

Comment (TheKodeToad) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: why do you want the old value?

Comment (moehreag) [reply 1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/HypixelMessages.java]: we don't, we want the potentially new value (the map of language to actual value) to add the value to it. The old value may potentially be `null`. This is the reason for a separated `.get` call.

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**